core/src/test/java/org/apache/iceberg/rest/responses/TestListNamespacesResponse.java

```diff
@@ -34,7 +34,7 @@ public class TestListNamespacesResponse extends RequestResponseTestBase<ListName
 
   @Test
   public void testRoundTripSerDe() throws JsonProcessingException {
-    String fullJson = "{\"namespaces\":[[\"accounting\"],[\"tax\"]]}";
+    String fullJson = "{\"namespaces\":[[\"accounting\"],[\"tax\"]],\"next-page-token\":null}";
```

Comment:
Older clients won't send this field, so I think we want to keep how the JSON was here for backward compatibility.

Reply:
One issue though is that since we have changed the `ListNamespaceResponse` to now include `next-page-token` which if not set by the server, will be `null`, and included in the `toString` representation.

```
  public String nextPageToken() {
    return nextPageToken;
  }

  @Override
  public String toString() {
    return MoreObjects.toStringHelper(this)
        .add("namespaces", namespaces())
        .add("next-page-token", nextPageToken())
        .toString();
  }

```
Thus when running these `testRoundTripSerDe` tests they will fail, as they will have the `next-page-token` in them.
```
Expected :"{"identifiers":[{"namespace":["accounting","tax"],"name":"paid"}]}"
Actual   :"{"identifiers":[{"namespace":["accounting","tax"],"name":"paid"}],"next-page-token":null}"
```

Reply:
`toString()` shouldn't matter when the object is serialized to JSON. I guess this is the downside of serializing to JSON via reflection (since we don't have a separate parser for this class to better control what we serialize/deserialize)

Reply:
Yea I think i see what you mean now, the test calls this at the end
```
  protected void assertRoundTripSerializesEquallyFrom(String json, T expected)
      throws JsonProcessingException {
    // Check that the JSON deserializes into the expected value;
    T actual = deserialize(json);
    assertEquals(actual, expected);

    // Check that the deserialized value serializes back into the original JSON
    Assertions.assertThat(serialize(expected)).isEqualTo(json);
  }
```

The issue seems to be occuring at this `Assertions.assertThat(serialize(expected)).isEqualTo(json);` since when we call the `serialize(expected)`  it does not go back to the original string json `"{"identifiers":[{"namespace":["accounting","tax"],"name":"paid"}]}"`  and instead takes the shape of the ListNameSpacesResponse(which now has the nextPageToken field).

Im not sure what we can do here besides either modify the json the test expects, or we need to to add another ListNameSpacesResponse class that contains a page token field(which seems not ideal)?

Reply:
for now I think we don't need to do anything other than adding `"next-page-token":null` to the expected json (as you've already done that).